README.md

```diff
@@ -27,31 +29,137 @@ npm install @dojo/widgets
+
+- All widgets are fully themeable.
+In a future release we will ship with several official Dojo 2 themes that can be applied to the widgets.
+
+- All widgets support internationalization (`i18n`)
+
+## Widgets
+
+### Form widgets
+[Button](src/button/README.md)
 
-Dojo 2 widgets are fully accessible (`a11y`), out of the box.
+[Checkbox/Switch](src/checkbox/README.md)
 
-Dojo 2 widgets are fully themeable. In a future release we will ship with several official Dojo 2 themes that can be applied to the widgets (all themes will live in a separate repo).
+[ComboBox](src/combobox/README.md)
 
-Dojo 2 widgets are fully internationalisable (`i18n`)
+[Label](src/label/README.md)
+
+[Radio](src/radio/README.md)
+
+[Select](src/select/README.md)
+
+[Slider](src/slider/README.md)
+
+[TextArea](src/TextArea/README.md)
+
+[TextInput](src/TextInput/README.md)
+
+### Layout widgets
+[SlidePane](src/slidepane/README.md)
+
+[TabPane](src/tabpane/README.md)
+
+### Misc widgets
+[Dialog](src/dialog/README.md)
 
 ## Conventions
 
+### EventHandlers
+
+You can register event handlers that get called when the corresponding events occur, by passing the handlers into a widget, in their `properties`.
+The naming convention for event handlers is as follows:
+
+- if the parent of the widget has the power to decide *if* an event is successful, i.e. can cancel the event, then the child widget will call an event handler in the following format:
+
+`onRequest[X]`, e.g. for a `close` event, the event handler called by the child widget must be called `onRequestClose`
+
+Here the child widget is requesting that the `close` event take place.
+
+- for events that will occur regardless of child/parent interaction, then the `Request` naming convention is dropped:
+
+`on[X]`, e.g. for a `dismiss` event, then event handler called by the child widget must be called `onDismiss`
+
+### Controlled vs. uncontrolled form widgets
+Currently, all of our *form* widgets are controlled widgets.
+This means that a parent widget is responsible for passing in the current value of the child widget.
+
+e.g.
+
+```ts
+[in WidgetFoo.ts]
+
+w(TextInput, {value: 'Hello Dojo!'}, []);
+```
+
+Here, `WidgetFoo` is the parent widget, which in turn, creates a child widget of type `TextInput`.
+In this instance, the `TextInput` is controlled by `WidgetFoo`, therefore the `TextInput` is known as a 'controlled' widget.
+Controlled form widgets have their current value passed into themselves by a parent widget. As well as their value, the parent is also responsible for determining other parts of the widget's state, such as if the widget is mandatory in the form, and also if the current value is valid.
+
+Here is how you would set the current value to 'Hello Dojo!', but also make the `TextInput` mandatory and invalid:
+
+```ts
+const props = {
+  value: 'Hello Dojo!',
+  invalid: true,    //defaults to false
+  required: true    //defaults to false
+}
+w(TextInput, props, []);
```

Comment:
Personally don't think it's needed, could potentially confuse people into thinking that they *need* to pass an empty array.